File: JavaScript/Tripletrouble.js
/*
tripledouble(451999277, 41177722899) == 1 // num1 has straight triple 999s and num2 has straight double 99s
tripledouble(1222345, 12345) == 0 // num1 has straight triple 2s but num2 has only a single 2
tripledouble(12345, 12345) == 0
tripledouble(666789, 12345667) == 1
*/

function tripledouble(num1, num2) {
  for (let i = 0; i <= 9; i++) {
    const digitStr = i.toString();
    if (num1.toString().includes(digitStr.repeat(3))) {
      if (num2.toString().includes(digitStr.repeat(2))) {
        return 1;
      }
    }
  }
  return 0;
}

describe('Initial Tests', () => {
  Test.assertEquals(tripledouble(451999277, 41177722899), 1);
  Test.assertEquals(tripledouble(1222345, 12345), 0);
  Test.assertEquals(tripledouble(12345, 12345), 0);
  Test.assertEquals(tripledouble(666789, 12345667), 1);
  Test.assertEquals(tripledouble(10560002, 100), 1);
  Test.assertEquals(tripledouble(1112, 122), 0);
});

File: C#/TripleTrouble.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;

public class Kata
{
  public static int TripleDouble(long num1, long num2)
  {
    string s1 = num1.ToString();
    string s2 = num2.ToString();
    int straightTriple = -1;

    for (int i = 0; i < s1.Length - 2; i++)
    {
      if (s1[i] == s1[i + 1] && s1[i] == s1[i + 2])
      {
        straightTriple = s1[i] - '0';
        break;
      }
    }

    if (straightTriple == -1)
        return 0;

    for (int i = 0; i < s2.Length - 1; i++)
    {
      if (s2[i] == s2[i + 1])
      {
        int straightDouble = s2[i] - '0';
        if (straightDouble == straightTriple)
          return 1;
      }
    }

    return 0;
  }
}

[TestFixture]
public class Tests
{
  [Test]
  [TestCase(451999277, 41177722899, ExpectedResult=1)]
  [TestCase(1222345, 12345, ExpectedResult=0)]
  [TestCase(12345, 12345, ExpectedResult=0)]
  [TestCase(666789, 12345667, ExpectedResult=1)]
  [TestCase(10560002, 100, ExpectedResult=1)]
  [TestCase(1112, 122, ExpectedResult=0)]
  public static int FixedTest(long s1, long s2)
  {
    return Kata.TripleDouble(s1, s2);
  }
}